In [3]:
import pandas as pd
import plotly.express as px

In [4]:
df = pd.read_csv(r'F:\GUVI_DATA_SCIENCE\Project\CarDekho-Used-Car-Price-Prediction\Datasets\Final\processed_final_data.csv')

In [12]:
df.sample()

manufacturer  model_year body_type transmission_type fuel_type  \
8078         Ford        2017     Sedan            Manual    Petrol   

      total_kms  total_owners     price insurance_type  mileage(kmpl)  ...  \
8078      71762             2  335000.0  Comprehensive          18.16  ...   

      torque(nm)  seats  number_of_cylinders  valves_per_cylinder  \
8078       112.0    5.0                    4                    4   

      super_charger turbo_charger drive_type steering_type cargo_volume  \
8078             No           Yes        FWD         Power        359.0   

      location  
8078   Kolkata  

[1 rows x 21 columns]

##### **Pearson Correlation**

In [14]:
numerical_variables = ['total_kms','price' ,'mileage(kmpl)' ,'engine(CC)', 'cargo_volume']

corr_df = df[numerical_variables]

corr_data = corr_df.corr()
corr_data

total_kms     price  mileage(kmpl)  engine(CC)  cargo_volume
total_kms       1.000000 -0.073208      -0.013896    0.091404      0.003125
price          -0.073208  1.000000      -0.249848    0.532672      0.006833
mileage(kmpl)  -0.013896 -0.249848       1.000000   -0.520890     -0.030764
engine(CC)      0.091404  0.532672      -0.520890    1.000000      0.015599
cargo_volume    0.003125  0.006833      -0.030764    0.015599      1.000000

In [15]:
fig = px.imshow(corr_data, text_auto = True)
fig.show()

- ##### Engine(CC) seems to have a postive correlation of 0.53 with the target variable (Price)